core/services/relay/relay.go

```diff
@@ -88,3 +101,28 @@ func (r *relayerAdapter) HealthReport() map[string]error {
 	maps.Copy(r.RelayerExt.HealthReport(), hr)
 	return hr
 }
+
+type ErrLogFiltersNotSupported struct {
+	relayName string
+}
+
+func (e ErrLogFiltersNotSupported) Error() string {
+	return fmt.Sprintf("Log filtering is not supported by relay %s", e.relayName)
+}
+
+func (r *relayerAdapter) OnCreateJob(rargs types.RelayArgs, q pg.Queryer) (err error) {
+	relay, ok := r.Relayer.(LogPollerCapable)
+	if !ok {
+		return nil
+	}
+	return relay.RegisterLogFilters(rargs, q)
```

Comment:
Why change the name? I understand if the other chains have no implementation yet, but the log poller is still just an implementation detail of the evm relay.
```suggestion
	return relay.OnCreateJob(rargs, q)
```

Reply:
Yeah, I guess that makes more sense.  I think I was trying to model it after `RelayerExt`, which includes methods that are specific to blockchains with a `ChainSet`.   The `LogPollerCapable` interface would be specific to blockchains which have logpoller, but I guess as we discussed this would probably only ever be evm.  And come to think of it, the evm relay should still have a `CreateJob()` method anyway... in case someone needs to add any actions not related to log poller.

Reply:
These extensions and adapters are meant to bridge the legacy in-process Relayer to the loop Relayer. It will never actually be possible to type cast the loops to see how they are implemented. You would always just see an opaque grpc client underneath that sends method calls over the wire.